src/status_im2/contexts/add_new_contact/events_test.cljs

```diff
@@ -10,47 +13,160 @@
+
+   ukey          {:user-public-key user-ukey
+                  :input           ukey
+                  :id              ukey
+                  :type            :public-key
+                  :public-key      ukey
+                  :state           :invalid
+                  :msg             :t/not-a-chatkey}
+
+   ens           {:user-public-key user-ukey
+                  :input           ens
+                  :id              ens
+                  :type            :ens
+                  :ens             ens-stateofus-eth
+                  :state           :resolve-ens}
+
+   (str " " ens) {:user-public-key user-ukey
+                  :input           (str " " ens)
+                  :id              ens
+                  :type            :ens
+                  :ens             ens-stateofus-eth
+                  :state           :resolve-ens}
+
+   ckey          {:user-public-key user-ukey
+                  :input           ckey
+                  :id              ckey
+                  :type            :compressed-key
+                  :state           :decompress-key}
+
+   link-ckey     {:user-public-key user-ukey
+                  :input           link-ckey
+                  :id              ckey
+                  :type            :compressed-key
+                  :state           :decompress-key}
+
+   link-ens      {:user-public-key user-ukey
+                  :input           link-ens
+                  :id              ens
+                  :type            :ens
+                  :ens             ens-stateofus-eth
+                  :state           :resolve-ens}))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; event handler tests (set-new-identity, set-new-identity-success/error)
+
+(def db
+  {:multiaccount             {:public-key user-ukey}
+   :networks/current-network "mainnet_rpc"
+   :networks/networks        {"mainnet_rpc"
+                              {:id     "mainnet_rpc"
+                               :config {:NetworkId 1}}}})
+
+(deftest empty-string-test ; no callbacks involved
+  (is (= (core/set-new-identity {:db db} "") {:db db})))
+
+(deftest uncompressed-key-test ; no callbacks involved
+  (is (= (core/set-new-identity {:db db} ukey)
+         {:db (assoc db
+                     :contacts/new-identity
+                     (core/init-contact
+                      {:user-public-key user-ukey
+                       :input           ukey
+                       :id              ukey
+                       :type            :public-key
+                       :public-key      ukey
+                       :state           :invalid
+                       :msg             :t/not-a-chatkey}))})))
+
+(deftest compressed-key-and-ens-test
+  (with-redefs [core/cb (fn [& args] args)] ; just args (no dispatching)
+    (let [;; event: compressed-key ->
+          db1 (core/set-new-identity {:db db} ckey)
+          c1  (core/init-contact
```

Comment:
I know this is just a test but still it would be nice to just use full variable names. e.g contact1 or contact-1

Reply:
@J-Son89 Here's my thinking here regarding variable names: if there are just a couple of key variables in a function, then I tend to make them very small, like abbreviations which are commonly understood placeholders for the longer names. In this case, reading this code, I might wonder what `c1` was initially, then see the first line is `init-contact` so I'd associate `c1` with `contact`. I did see some other places in the codebase that this abbreviation is used. WDYT? I can change it if you feel strongly about it, no worries.